### PR TITLE
fix: filters enhance

### PR DESCRIPTION
**Pull request by sagarnaikjuspay**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by Riddhiagrawal001 — DISMISSED** [review]

File: src/screens/HyperSwitch/HSwitchRemoteFilter.res [hunk 1]
```diff
@@ -207,18 +207,14 @@ module RemoteTableFilters = {
...
-      ~updateExistingKeys,
-      ~startTimeFilterKey,
-      ~endTimeFilterKey,
-    )
-
     let customViewTop = <SearchBarFilter placeholder setSearchVal searchVal />
 
     React.useEffect0(() => {
-      setInitialFilters()
+      if filterValueJson->Js.Dict.keys->Js.Array2.length <= 0 {
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/HSwitchRemoteFilter.res:214]: you can just check for empty dict

File: src/components/DateRangePicker.res [hunk 2]
```diff
@@ -540,7 +540,9 @@ module Base = {
     }
 
     let buttonText = {
-      showTime
+      startDateVal->Js.String2.length === 0 && endDateVal->Js.String2.length === 0
```
Comment (PritishBudhiraja) [inline src/components/DateRangePicker.res:543]: can we start using the rescript core changes from every PR?
For Reference - [Conversions](https://github.com/rescript-association/rescript-core/blob/main/migration/migration.toml)

File: src/screens/HyperSwitch/HSwitchRemoteFilter.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/HSwitchRemoteFilter.res:214]: Same can we use the conversions.

File: src/screens/HyperSwitch/HSwitchRemoteFilter.res [hunk 3]
```diff
@@ -207,18 +207,14 @@ module RemoteTableFilters = {
...
+        setFilters(_ => Js.Dict.empty()->Some)
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/HSwitchRemoteFilter.res:215]: Same conversion.

**Review by PritishBudhiraja — CHANGES_REQUESTED** [review]

File: src/screens/HyperSwitch/HSwitchRemoteFilter.res [hunk 4]
```diff
@@ -207,18 +207,14 @@ module RemoteTableFilters = {
...
-      ~startTimeFilterKey,
-      ~endTimeFilterKey,
-    )
-
     let customViewTop = <SearchBarFilter placeholder setSearchVal searchVal />
 
     React.useEffect0(() => {
-      setInitialFilters()
+      if filterValueJson === Dict.make() {
+        setFilters(_ => Js.Dict.empty()->Some)
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/HSwitchRemoteFilter.res:215]: Dict.make()

**Review by PritishBudhiraja — APPROVED** [review]

File: src/components/DateRangePicker.res [hunk 5]
```diff
@@ -540,7 +540,9 @@ module Base = {
     }
 
     let buttonText = {
-      showTime
+      startDateVal->String.length <= 0 && endDateVal->String.length <= 0
```
Comment (Riddhiagrawal001) [inline src/components/DateRangePicker.res:543]: We can directly write `startDateVal->String.length === 0 && endDateVal->String.length === 0`

File: src/screens/HyperSwitch/HSwitchRemoteFilter.res [hunk 6]
```diff
@@ -179,16 +179,13 @@ module RemoteTableFilters = {
...
-      ~updateExistingKeys,
-      ~startTimeFilterKey,
-      ~endTimeFilterKey,
-    )
-
     let customViewTop = <SearchBarFilter placeholder setSearchVal searchVal />
 
     React.useEffect0(() => {
-      setInitialFilters()
+      if filterValueJson->Js.Dict.keys->Js.Array2.length === 0 {
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/HSwitchRemoteFilter.res:185]: Please use core functions for Js.Dict.keys and Js.Array2.length

File: src/screens/HyperSwitch/HSwitchRemoteFilter.res [hunk 7]
```diff
@@ -227,7 +224,7 @@ module RemoteTableFilters = {
     let filterData = filterDataJson->Belt.Option.getWithDefault(Dict.make()->Js.Json.object_)
 
     React.useEffect1(() => {
-      if filterValueJson->Dict.keysToArray->Array.length > 0 {
+      if filterValueJson->Js.Dict.keys->Js.Array2.length != 0 {
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/HSwitchRemoteFilter.res:227]: Please use core functions for Js.Dict.keys and Js.Array2.length

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**